data/cards-expansion_set-shatteredAlliances.cfg

```diff
@@ -0,0 +1,166 @@
+		portrait_focus_height: 120,
+		artist: "wendyyoon",
+		cost: 4,
+		school: "@eval GOLD",
+		possible_targets: "def(class game_state game, class creature avatar, [Loc] targets) ->[Loc]|null
+		   if(creature != null and targets = [],
+					filter(game.players[avatar.controller].summoning_locs(game, creature),
+                    in_side where in_side = (value.x = 0 or value.x+1 = game.map_dimensions.x)
+                    )
+                  )",
+		creature: {
+			abilities: ['Swift'],
+			tags: ['Mounted'],
+			attack: 5,
+			life: 4,
+			game_sounds: "soldier_blade",
+			sound_swish: "medium",
+			sound_impact: "wood",
+			
+			activated_abilities: [
+				{
+					name: 'Maneuver',
+					cost: 1,
+					portrait: "kings-rider.png",
+					school: "@eval GOLD",
+					rules: "Spend 1 movement to move to an adjacent tile.",
+					once_per_turn: false,
+					uses_movement: true,
+
+					possible_targets: "
+					 def(class game_state game, class creature creature, [Loc] targets) ->[Loc]|null
+						if(targets = [], filter(
+								game.get_adjacent_locs(creature.loc),
+							game.is_loc_on_board(value) and
+							game.creature_at_loc(value) = null), null)
+					",
+
+					on_play: "def(class game_state game, class message.play_card info) ->commands [
+						creature.set_loc(game, info.targets[0]),
+                        add(creature.spaces_moved_this_turn, 1)
+					] asserting size(info.targets) = 1
+					  where creature = info.get_creature(game)",
+				}
+			]
+
+		}
+	},
+	
+	"Junk wagon": {
+		name: "Junk wagon",
+		type: "creature",
+        set: "shatteredAlliances",
+		portrait: "wall-of-stone.png",
+		portrait_focus: [429,236],
+		portrait_focus_height: 110,
+		cost: 3,
+		loyalty_cost: 1,
+        rarity: 0,
+		school: "@eval MATERIA",
+		creature: {
+			tags: ['Mechanical'],
+			abilities: ['Valiant'],
+			attack: 4,
+			life: 1,
+			triggered_abilities: [{
+				name: "Baricade ",
+				rules: "When Junk Wagon dies, an (Materia) Baricade Land is summoned in the lane it was in unleas there is another land on it already.",
+				on_die: "def(class creature creature, class game_state game) ->commands
+				if(not game.land_at_loc(creature.loc),
+				game.summon_land(lib.citadel.create_creature('Baricade', { controller: creature.controller, loc: creature.loc, card_name: 'Baricade' }))
+				)",
+			}],
+		},
+	},
+
+//Entropia cards
+	
+	"Mindless Mob": {
+		name: "Mindless Mob",
+		set: "shatteredAlliances",
+		rarity: 0,
+		type: "creature",
+		portrait: "kings-rider.png",
+		portrait_focus: [451,198],
+		portrait_focus_height: 120,
+		artist: "wendyyoon",
+		cost: 4,
+		school: "@eval BLOOD",
+		possible_targets: "def(class game_state game, class creature avatar, [Loc] targets) ->[Loc]|null
+		   if(creature != null and targets = [],
+					filter(game.players[avatar.controller].summoning_locs(game, creature),
+                    in_center where in_center = (game.map_dimensions.x%2 = 1 and value.x*2+1 = game.map_dimensions.x)
```

Comment:
Now I can't recall of any map in the game being even-numbered of lanes. I would not take for granted an odd number of lanes. But it is OK.

Reply:
This was designed with the multiplayer board in mind. If campaigns can have custom rules then we wont be able to foresee those rules while designing cards. :( I would say it will have to be on the campaign creators to check whether cards will function after their customization. 

Can we forbid cards from the campaign, with e.g. tags or something? [not preferred]
Can we set up in the campaign scenarios that such and such scenario cant be played when player has X card in deck?

Reply:
> [...] If campaigns can have custom rules then we wont be able to foresee those rules while designing cards. [...]

I think cards development (and even cards balancing) should be very close to lore creating, as much as campaign development. But you are right, cards development better abstains from foreseeing campaign development while possible.

> Can we forbid cards from the campaign, with e.g. tags or something? [not preferred]

I agree we don't do that.

> Can we set up in the campaign scenarios that such and such scenario cant be played when player has X card in deck?

I don't see there is enough a need for doing this, at least about this case.

> Maybe add a fail-safe for the even number board to place it on the "right-middle" lane.

I don't think this is necessary either, because your code already allows placing `Mindless Mob` in one of the middle lanes of maps with even number of lanes (assuming you can have such maps at all).

Reply:
I did some testing with maps having even number of lanes. These are my first opinions on what happened.

* 2x6 map with two villages: The two lanes are too much far away one from the other. Besides this, I think it is playable. But not release-ready.

* 4x4 map: Two seals do not show, so absolutely not release-ready.

* 6x6 map with 6 villages: Mostly playable, but maybe the game is not prepared for it yet. Played a game against AI. The AI does pretty well out of the box... but I think it suffers too much from shadow deck damage. Not release-ready to my opinion.

* 6x5 map with 6 villages: Mostly playable, but some own seals get covered with the cards in own hand, so not release-ready. The game crashed but I didn't read the log.

* 4x5 map with 6 villages: Some enemy seals get covered with the cards in opponent's hand, so not release-ready.

* Standard (5 lanes, 19 tiles) map stripped from one of their outermost lanes: Can't get this to render OK. Not release-ready.

I think some combinations are playable, but since none renders actually perfect, one could say let's fix this later 😃.

Reply:
@pacnutek the idea for sets is that you can create an expansion set or a complete set. A complete set would have its own campaign(s) where the cards are earned and used. 

"Currently the only standalone set is 'core' which is under Dave as the head of the project. This set will be the main trunk of the game. Because we wish to focus our limited efforts and play testing resources into polishing this core set, it is not encouraged for anyone to start working on another standalone set until the game is released. Nevertheless, feel encouraged to make cards, test and try things out, and experiment."

You can read more here: https://github.com/davewx7/citadel/wiki/A-brief-explanation-of-the-set-system

Reply:
Yeah, @pacnutek is proposing a set of four materia cards an an entropia card. The way I see it, expansion sets should be school-neutral, and this wouldn't be. If you have no plans for other cards, they would apply in the outer bound of the core set, outside, or in the big experimental set,  were @sapientN3T is writing his cards.

Reply:
@sapientN3T: I read that article earlier, it was this very article actually that gave me the idea of creating my own set :) It encourages the reader to do so, more than once. Just saying :]

As for it being school-neutral - you do realize it is work in progress, right? I have ideas and plans for more cards, it was just the Materia ones that where the easiest to implement. Following the Agile ideas of "Fail fast, fail often to learn fast and learn often" I gave you this piece of code to get this wonderful feedback :D 
I dont know if I will create a campaign scenario for it or not. Time will tell. The idea of a OSS is that everyone can help so I am counting on someone to help me too :) To be honest I never planned to create my own set, it is just that there is nowhere in the documentation, at least nowhere I have looked, that everyone can pitch in into the Experimental set. I dont like the idea that a freshman like me will mess someone elses code. I didnt want to name mine Experimental 2 so I named it "parasite lost" at the beginning ;) I is no problem for me to move the code to the experimental set. My aim is more that, now I know it is Dave, someone who administers the core set has as much variety to choose from. And maybe, just maybe one of my proposals will get into the final product (after they get hammered under the balancing hammer of course).